### ISSUE TITLE
Decide: Does demo site audience include superadmins or not?

### ISSUE THREAD

**Issue by atz** [projecthydra-labs/hyku#728]
Pros:
- Exposes the fullest range of features/data for evaluation (much of the novel impressive stuff we are building is at this level)

Cons:
- User interactions will be potentially destructive and surprising.  
- Greater abuse potential.

If decided in favor, the task will be to complete a feature to allow self-promotion.  
If decided against, the task will be to ensure security of acct. creation page.  

In either case, we may need configurability to present the demo differently than the default.  So this ticket is not asking: "what should hyku do by default?"  It is asking: "what should the demo site do?"

**Comment by mjgiarlo** [comment]
Adding another dimension to @atz's question: we should have answers for MVP and post-MVP. Let's discuss on today's standup.

**Comment by mjgiarlo** [comment]
This is for post-MVP.

**Comment by mjgiarlo** [comment]
Discussing with @hannahfrost @bbranan on Friday's manager meeting. We feel this should be driven by HykuDirect's needs.

**Comment by hannahfrost** [comment]
Please see this articulation/breakdown: https://docs.google.com/document/d/10SN-IqPWpnZElmLyXuvIpi8tqBco0V4BilGUwhvM1Qo/edit

**Comment by atz** [comment]
We also discussed providing a mechanism that exposes superadmin features to us (but not the public), so that we have some ability to review/approve those features.

**Comment by mjgiarlo** [comment]
Talked with @hannahfrost  @bbranan and other managers on Friday and decided that the demo site audience should include superadmins. We'd like to include the following in the MVP scope:

* Need a way to make sure every instance has at least one superadmin (and, ideally, superadmins should have the ability to create other superadmins)
* Tenant creation should be restricted to superadmins, and should include a step where they grant access to one or more tenant admin users 

I suggest we discuss this from a technical design standpoint in our Monday kickoff meeting when all the stakeholders (managers, engineers) are available.

Thanks for writing this up, @atz.

**Comment by jcoyne** [comment]
Does this mean we are removing the ability for people to create their own repositories from the demo site?

**Comment by mjgiarlo** [comment]
@jcoyne Yes, for most people but not us; I would expect all of us in this thread to be made superadmins of the demo site.

**Comment by jcoyne** [comment]
@mjgiarlo sounds good. Thanks for the clarification.

**Comment by mjgiarlo** [comment]
@atz Edited my comment above to clarify. We'll revisit this post-MVP.

**Comment by atz** [comment]
@mjgiarlo: are we tracking tracking the outstanding tasks in this ticket?

**Comment by mjgiarlo** [comment]
@atz Ah, no -- I'll make sure those go into new issues.

**Comment by mjgiarlo** [comment]
@atz See #733, #725, #734, and #735.

**Comment by mjgiarlo** [comment]
@hannahfrost @atz @jcoyne I decomposed this ticket into three others (and @hannahfrost created a fourth), that should be done in this order:

* #733 -- grant superadmin role to users in the demo instance. devops ticket, can be done in Rails console on AWS until we have a better mechanism for this.
* #725 -- superadmins should have a UI for creating tenants (incl. granting 1+ admin users initial access to the tenant)
* #735 -- restrict tenant creation to superadmins/disable self-provisioning

Closing.
